src/ipc/cancel-speak.test.ts

```diff
@@ -0,0 +1,44 @@
+import { IpcMain, IpcMainEvent } from 'electron';
+import { fakeIpc } from '../../test/ipc';
+import mockOf from '../../test/mockOf';
+import deferred from '../../test/deferred';
+import exec from '../utils/exec';
+import register, { channel } from './cancel-speak';
+
+const execMock = mockOf(exec);
+
+jest.mock('../utils/exec');
+
+test('cancel-speak', async () => {
+  // Register our handler.
+  const handle = jest.fn<
+    ReturnType<IpcMain['handle']>,
+    Parameters<IpcMain['handle']>
+  >();
+  register({ handle } as unknown as IpcMain);
+
+  // Things should be registered as expected.
+  expect(handle).toHaveBeenCalledWith(channel, expect.any(Function));
+
+  // Is the handler wired up right?
+  const [, handler] = handle.mock.calls[0];
+  await handler({} as IpcMainEvent);
+
+  expect(execMock).toHaveBeenCalledWith('spd-say', ['-C']);
+});
+
+test('does not duplicate request to cancel speech if job already exists', async () => {
+  const { ipcMain, ipcRenderer } = fakeIpc();
+  register(ipcMain);
+
+  const { promise, resolve } = deferred<{ stdout: string; stderr: string }>();
+  execMock.mockReturnValueOnce(promise);
+
+  const firstCancel = ipcRenderer.invoke(channel);
+  const secondCancel = ipcRenderer.invoke(channel);
+  resolve({ stdout: '', stderr: '' });
+  await firstCancel;
+  await secondCancel;
+
+  expect(execMock).toBeCalledTimes(1);
+});
```

Comment:
unclear why that's necessary.

Reply:
I have implemented the `cancelSpeak` API so it only makes once call to `spd-say -C` at a time:

https://github.com/votingworks/kiosk-browser/pull/130/commits/bca0c9787833eb3bccf2ab61a470134c664184fa#diff-c86eb634d31dc295068899900e61791c1d8bf736e08714e6d362f1282fee20bdR23-R40

We call `mute` 2 or 3 times in succession on app start. That results it `spd-say` exiting with code 1. We could just allow those processes to fail. But because it seemed messy, I have `kiosk-browser` taking care not to call twice in a row. The test is meant to check that the behavior is working.

Does that check out?

Reply:
I'm a little wary of any "smart" middle layer like this that, one day, could drive us crazy as we wonder why calls are being de-duped. I *think* it's probably okay here, and the risk is low, so I'm good with it.